shop-ui/src/main/java/com/maxshop/viewModel/CategoriesViewModel.kt

```diff
@@ -3,59 +3,63 @@ package com.maxshop.viewModel
 import androidx.lifecycle.MutableLiveData
 import com.example.internetshop.presentation.utils.SingleLiveEvent
 import com.maxshop.mapper.CategoryMapper
+import com.maxshop.model.RecyclerItem
 import com.maxshop.usecase.GetCategoriesUseCase
 import com.maxshop.viewState.CategoryViewState
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
-import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 class CategoriesViewModel @Inject constructor(
     private val getCategoriesUseCase: GetCategoriesUseCase,
     private val categoryMapper: CategoryMapper
 ) : BaseViewModel() {
 
-    val events = SingleLiveEvent<CategoryEvents>()
-    val categoriesLiveData = MutableLiveData<List<CategoryViewState>>()
+    val events = SingleLiveEvent<Event>()
+    val categories = MutableLiveData<List<RecyclerItem>>()
     val progressBar = MutableLiveData<Boolean>()
 
     fun getProducts() {
         getCategoriesUseCase.execute()
-            .timeout(60, TimeUnit.SECONDS)
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .doOnSubscribe { progressBar.value = true }
-            .doFinally {
-                progressBar.value = false
-            }
-            .subscribe(
-                { it ->
-                    categoriesLiveData.value = it.map {
-                        categoryMapper.toViewState(it).also {
-                            compositeDisposable.add(
-                                it.events.subscribe {
-                                    when (it) {
-                                        is CategoryViewState.Event.OnProductClick -> {
-                                            events.value =
-                                                CategoryEvents.OpenCategoryProductListEvents(it.name)
-                                        }
+            .doFinally { progressBar.value = false }
+            .map {
+                it.map {
+                    categoryMapper.toViewState(it).also {
+                        compositeDisposable.add(
+                            it.events.subscribe {
+                                when (it) {
```

Comment:
Please create a method `onCategoryViewStateEvent(event) and handle events from the view state in this method. This will increase readability of `getProduct` method